python_modules/dagster/dagster/_core/definitions/decorators/asset_check_decorator.py

```diff
@@ -0,0 +1,117 @@
+from functools import update_wrapper
+from typing import Any, Callable, Mapping, Optional, Set, Union
+
+from dagster import _check as check
+from dagster._annotations import experimental
+from dagster._config import UserConfigSchema
+from dagster._core.definitions.asset_check_result import AssetCheckResult
+from dagster._core.definitions.asset_check_spec import AssetCheckSpec
+from dagster._core.definitions.asset_checks import AssetChecksDefinition
+from dagster._core.definitions.assets import AssetsDefinition
+from dagster._core.definitions.decorators.asset_decorator import (
+    asset_key_from_coercible_or_definition,
+    build_asset_ins,
+)
+from dagster._core.definitions.events import CoercibleToAssetKey
+from dagster._core.definitions.output import Out
+from dagster._core.definitions.policy import RetryPolicy
+from dagster._core.definitions.source_asset import SourceAsset
+
+from .op_decorator import _Op
+
+AssetCheckFunctionReturn = AssetCheckResult
+AssetCheckFunction = Callable[..., AssetCheckFunctionReturn]
+
+
+@experimental
+def asset_check(
+    *,
+    asset: Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset],
```

Comment:
@schrockn @alangenfeld @jamiedemaria - thoughts on this parameter name? Vs. e.g. `asset_key`? I imagine that whatever we choose here is the same name that we'd want to use on `AssetDependency` (or whatever we choose to name it).

Reply:
riffing 
* `dep` (`deps` is whats currently on `AssetSpec`) 
* `target`
* `for`
* `for_asset` 
* `on`

Reply:
Also, it might be good to be consistent with the parameter on `AssetCheckSpec` (currently named `asset_key`).

Reply:
`deps` feels a bit wrong for this use case. It would imply that the asset_check has data/ordering dependencies on some asset(s) before it can run, when instead we're trying to indicate that this `@asset_check` checks a specific asset. 

To use the graph analogy - `deps` defines an edge between two nodes (assets). Based on my understanding we're trying to attach the `asset_check` as like a decorator on the corresponding asset node, not as a node itself with an edge to the corresponding asset


i like `target` and `for_asset`, but that would be introducing new vocab (less so for `for_asset`). `asset` would also be fine. I think `asset_key` too heavily implies it should be an `AssetKey` type, especially once `AssetSpec`s are in play 

+1 that it should be the same for `AssetCheckSpec`

Reply:
I think `asset` or`for_asset` are the best options. Agree that `asset_key` is undesirable for the reasons you stated @jamiedemaria